File: Patrones Creacionales/Factory Method/Ejemplo.js
class Product {
  operation() {}
}

class ConcreteProductA extends Product {
  operation() { 
    return "Producto A"; 
  }
}

class Creator {
  factoryMethod() {}
  create() { 
    return this.factoryMethod().operation(); 
  }
}

class CreatorA extends Creator {
  factoryMethod() { 
    return new ConcreteProductA(); 
  }
}

console.log(new CreatorA().create());

File: Patrones Creacionales/Factory Method/Ejemplo.py
class Product:
    def operation(self): pass

class ConcreteProductA(Product):
    def operation(self):
        return "Producto A"

class ConcreteProductB(Product):
    def operation(self):
        return "Producto B"

class Creator:
    def factory_method(self): pass
    def create(self):
        return self.factory_method().operation()

class CreatorA(Creator):
    def factory_method(self):
        return ConcreteProductA()

print(CreatorA().create())
